### PR TITLE
[RHINENG-1025] update class components to fix table

### PR DESCRIPTION
**Pull request by johnsonm325**
This PR updates a number of class components to functional components and further breaks up the drift table into more manageable components. Functionality should remain the same. The hope is that this will fix the column resizing issue on the drift table in stage/prod.

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: src/SmartComponents/DriftPage/DriftToolbar/__tests__/DriftToolbar.tests.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { Provider } from 'react-redux';
+import { shallow, mount } from 'enzyme';
```
Comment (mkholjuraev) [inline src/SmartComponents/DriftPage/DriftToolbar/__tests__/DriftToolbar.tests.js:3]: RTL should be used instead of enzyme.

**Review by mkholjuraev — COMMENTED** [review]
Tests that assert on the functional calls are not encouraged in RTL as they usually test implementation details rather than user behavior on the UI. I see that most of the new tests in this PR are testing implementation detail. Would you say we should also opt for tests on user behavior while we are on this? Reference: https://medium.com/@ankitsingh80741/a-guide-to-effective-testing-with-react-testing-library-475a3bf1988c

File: src/SmartComponents/DriftPage/DriftToolbar/__tests__/DriftToolbar.tests.js (hunk 1 above)
Comment (johnsonm325) [reply src/SmartComponents/DriftPage/DriftToolbar/__tests__/DriftToolbar.tests.js:3]: yes, but I didn't want to take time to convert this entire test file into rtl. that can be done later. I just did this to get a better snapshot temporarily.

**Review by adonispuente — APPROVED** [review]
Muslimjon has some comments to be addressed, but overall when running locally, no functionality seems to be affected. I wasnt able to encounter the issue locally (but I believe thats one of the main issues with it). 
Overall the refactor is great. Theres a point to be made with converting test but it doesnt all have to be done with this specific PR.

**Pull request merged**